packages/robo/src/cli/utils/cloudflared.ts

```diff
@@ -73,22 +152,72 @@ export function isCloudflaredInstalled(to = DEFAULT_BIN_PATH): boolean {
 	return fs.existsSync(to)
 }
 
+export async function initializeCloudflareTunnel(): Promise<boolean> {
+	if (!process.env.CLOUDFLARE_API_KEY || !process.env.CLOUDFLARE_ACCOUNT_ID || !process.env.CLOUDFLARE_DOMAIN) {
+		cloudflareLogger.warn('Missing required Cloudflare environment variables.');
+		return false;
+	}
+
+	try {
+		if (process.env.CLOUDFLARE_TUNNEL_ID) {
+			const roboTunnel = await cloudflareRequest(`/accounts/${process.env.CLOUDFLARE_ACCOUNT_ID}/cfd_tunnel/${process.env.CLOUDFLARE_TUNNEL_ID}/token`);
+			cloudflareLogger.debug(`Robo tunnel: ${JSON.stringify(roboTunnel)}`);
+
+			updateEnvFile('CLOUDFLARE_TUNNEL_ID', process.env.CLOUDFLARE_TUNNEL_ID);
+			updateEnvFile('CLOUDFLARE_TUNNEL_TOKEN', roboTunnel.result as string);
+		} else {
+			const newTunnel = await cloudflareRequest(
+				`/accounts/${process.env.CLOUDFLARE_ACCOUNT_ID}/tunnels`,
+				'POST',
+				{
+					config_src: "cloudflare",
+					name: 'robo',
+				}
+			);
+
+			const { id, token } = newTunnel.result as CFDTunnel;
+
+			cloudflareLogger.debug(`Created new tunnel: robo (${id})`);
+
+			updateEnvFile('CLOUDFLARE_TUNNEL_ID', id);
+			updateEnvFile('CLOUDFLARE_TUNNEL_TOKEN', token);
+
+			installService(token);
+		}
+
+		await updateTunnelConfig(process.env.CLOUDFLARE_TUNNEL_ID, process.env.CLOUDFLARE_ACCOUNT_ID);
+		return true;
+	} catch (error) {
+		cloudflareLogger.error('Failed to initialize Cloudflare tunnel:', error);
+		return false;
+	}
+}
+
 export function startCloudflared(url: string) {
+	const tunnelId = process.env.CLOUDFLARE_TUNNEL_ID;
+	const tunnelToken = process.env.CLOUDFLARE_TUNNEL_TOKEN;
```

Comment:
Can we reduce the number of items requested from the user by requesting the tunnel token from Cloudflare ourselves?

Reply:
Currently only the Account ID, API Key, and the domain are requested from the user. The tunnel ID and token are stored in the .env file after the initial tunnel creation for later use.